Workflow: added unit testing and e2e testing for login and logout

H-chai opened the pull request.

H-chai committed:
install Prettier as dev dependency

H-chai committed:
install ESLint

H-chai committed:
add .eslintrs.json to configure ESLint

H-chai committed:
add a "lint" task

H-chai committed:
Set up lint-staged for pre-commit checks

H-chai committed:
instruct lint-staged to a specific command for a specific file type

H-chai committed:
Create a save hook

H-chai committed:
Set to use Prettier as default formatter

H-chai committed:
add linting rules to auto fix errors on save

H-chai committed:
install jest and add scripts test and test-run

H-chai committed:
install eslint-plugin-jest to allow jest functions to be used in test

H-chai committed:
update the file to enable eslint-plugin-test

H-chai committed:
install babel

H-chai committed:
add babel.config.json file and update to ensure test can use modern syntax

H-chai committed:
add testing files for login and logout function

H-chai committed:
npx eslint --init

H-chai committed:
configured ESLint setup

H-chai committed:
add eslint.options

H-chai committed:
fix to the original one

H-chai committed:
Fix linting issues for Jest test files

H-chai committed:
Update env

H-chai committed:
add node to overrides: "evd"

H-chai committed:
delete console.log

H-chai committed:
change clear to removeItem

H-chai committed:
add file and test remove function

H-chai committed:
add file and test logout function

H-chai committed:
(Test) add console.log

H-chai committed:
remove console.log

H-chai committed:
install Cypress

H-chai committed:
update with configuration data for linting Cypress tests

H-chai committed:
add a new script for cypress

H-chai committed:
Opened Cypress and install configuration data in project

H-chai committed:
instruct Prettier to use single quotes instead of double quotes

H-chai committed:
Cypress test login form

H-chai committed:
Cypress test login form with invalid credentials

H-chai committed:
change to check if body has class name "logged-in" after login

H-chai committed:
Cypress test logout

H-chai committed:
Updated and test will fail because error message is not shown to user when try to login with invalid credentials

EliNygard commented:
Hi, 

from my knowledge this looks really good! I have not cloned and checked your tests, only read the files. Two questions :)

1. I see that you have both .eslintrc.json and eslint.config.mjs and it looks like the content in the .mjs file is also in the .json file. Just check and see if you can delete the .mjs file if it is a duplicate.
2. In your cypress tests you use the local host url in cy.visit. Will that work for another developer? Should it be a deployed site to refer to?

Best, Eli

H-chai committed:
Simplify the mockLocalStorage object to only include the methods used in tests

H-chai committed:
add removeItem: jest.fn

AnnaAaBrekke commented:
Hi! The setup looks good. I was wondering if you've thought about possibly adding testing (like unit tests) to the pre-commit stage as well. I'm not 100% sure if it's necessary though. 
That said, everything looks great to me, and you've done a really good job. :)

H-chai committed:
Refactor ESLint config to use flat config system, replacing .eslintrc.json with eslint.config.mjs

H-chai committed:
Install Ssss and compile scss, and update HTML with correct css file path

H-chai committed:
Update with deployed page URL instead of local host URL

H-chai commented:
> Hi,
> 
> from my knowledge this looks really good! I have not cloned and checked your tests, only read the files. Two questions :)
> 
> 1. I see that you have both .eslintrc.json and eslint.config.mjs and it looks like the content in the .mjs file is also in the .json file. Just check and see if you can delete the .mjs file if it is a duplicate.
> 2. In your cypress tests you use the local host url in cy.visit. Will that work for another developer? Should it be a deployed site to refer to?
> 
> Best, Eli

Thank you for the review, Eli!
1. I refactored ESLint config to use flat config system, deleted .eslintrc.json and replaced all to eslint.config.mjs.
2. I updated cypress tests file with deployed URL!

:)

EliNygard commented:
You are very welcome! Nice! 🦸‍♀️

(Good that you chose to use .js. I forgot to mention, I read that ESLint reccomends .js for the config file 🙏🏻)

H-chai committed:
Update Bootstrap Dark import to use CDN

H-chai committed:
update to force Cypress to type into the field

H-chai committed:
Delete comment out lines

H-chai committed:
modify lint-staged configuration to run npm run test-unit before commit

H-chai committed:
Testing pre-commit hook

H-chai committed:
Testing pre-commit hook

H-chai committed:
Add test for login without credentials

H-chai committed:
Add failing test to check pre-commit

H-chai committed:
back to the original one

H-chai committed:
husky npx mrm@2 lint-staged

H-chai commented:
> Hi! The setup looks good. I was wondering if you've thought about possibly adding testing (like unit tests) to the pre-commit stage as well. I'm not 100% sure if it's necessary though. That said, everything looks great to me, and you've done a really good job. :)

Hi Anna,

Thank you for the feedback.
I added "*.test.js": [ "npm run test-unit" ] in package.json!

H-chai committed:
updated second test text

H-chai committed:
Replaced env with globals (using the flat config system)
